api/src/main/java/me/lokka30/treasury/api/economy/EconomyProvider.java

```diff
@@ -140,15 +145,162 @@
+     * @param permissions  the permissions that the given player has to have on the {@link BankAccount account}
+     * @author MrNemo64
+     * @see #retrieveAllBankAccountsPlayerIsMemberOf(UUID, EconomySubscriber)
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    default void retrieveAllBankAccountsPlayerHasPermission(
+            @NotNull UUID playerId,
+            @NotNull EconomySubscriber<Collection<UUID>> subscription,
+            @NotNull BankAccountPermission @NotNull ... permissions
+    ) {
+        Objects.requireNonNull(playerId, "playerId");
+        Objects.requireNonNull(subscription, "subscription");
+        Objects.requireNonNull(permissions, "permissions");
+        EconomySubscriber.asFuture(this::retrieveBankAccountIds).exceptionally(throwable -> {
+            if (throwable instanceof EconomyException) {
+                subscription.fail((EconomyException) throwable);
+            } else {
+                subscription.fail(new EconomyException(FailureReason.OTHER_FAILURE, throwable));
+            }
+            return new HashSet<>();
+        }).thenAccept(uuids -> {
+            if (uuids.isEmpty()) {
+                subscription.succeed(uuids);
+                return;
+            }
+            Set<UUID> ret = new HashSet<>();
+            for (UUID uuid : uuids) {
+                EconomySubscriber.<BankAccount>asFuture(subscriber -> retrieveBankAccount(uuid, subscriber)).exceptionally(
+                        throwable -> {
+                            if (throwable instanceof EconomyException) {
+                                subscription.fail((EconomyException) throwable);
+                            } else {
+                                subscription.fail(new EconomyException(FailureReason.OTHER_FAILURE, throwable));
+                            }
+                            return null;
+                        }).thenCompose(account -> {
+                    if (account == null) {
+                        return CompletableFuture.completedFuture(false);
+                    }
+                    AtomicBoolean modified = new AtomicBoolean(false);
+                    EconomySubscriber.<Boolean>asFuture(subscriber -> account.isBankMember(playerId, subscriber)).exceptionally(
+                            throwable -> {
+                                if (throwable instanceof EconomyException) {
+                                    subscription.fail((EconomyException) throwable);
+                                } else {
+                                    subscription.fail(new EconomyException(FailureReason.OTHER_FAILURE, throwable));
+                                }
+                                return null;
+                            }).thenAccept(val -> {
+                        if (!val) {
+                            modified.set(true);
+                        }
+                    });
```

Comment:
I don't think we should check if the player is a memeber on this method, only if they have the permissions
A player will be a bank memeber as long as he has at least one permission on it so if they have the permissions they are bank members